### PR TITLE
Fixing Mac build (includes M1)

### PR DESCRIPTION
**Pull request by querolita**
This PR fixes the commands to build in Mac, both if it is x86_64 or arm64 architectures. It adds further instructions and clarifications. It also provides the required steps to make Ocaml Platform work properly in VSCode.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -71,7 +71,7 @@ libp2p_helper:
 
 genesis_ledger: ocaml_checks
 	$(info Building runtime_genesis_ledger)
-	ulimit -s 65532 && (ulimit -n 10240 || true) && env MINA_COMMIT_SHA1=$(GITLONGHASH) dune exec --profile=$(DUNE_PROFILE) src/app/runtime_genesis_ledger/runtime_genesis_ledger.exe -- --genesis-dir $(GENESIS_DIR)
+	((ulimit -s 65532 || true) || true) && (ulimit -n 10240 || true) && env MINA_COMMIT_SHA1=$(GITLONGHASH) dune exec --profile=$(DUNE_PROFILE) src/app/runtime_genesis_ledger/runtime_genesis_ledger.exe -- --genesis-dir $(GENESIS_DIR)
```
Comment (mimoo) [inline Makefile:74]: it'd be good to have someone from infra look at this change :D I admit I don't really know the consequences of these ulimit lines or even why they're here in the first place (seems odd to me)

File: README-dev.md [hunk 2]
```diff
@@ -85,10 +115,7 @@ let g:syntastic_ocaml_checkers=['merlin']
 - Now `/usr/bin/opam install merlin ocp-indent core async ppx_jane ppx_deriving` (everything we depend on, that you want autocompletes for) for doc reasons
 - Make sure you have `au FileType ocaml set omnifunc=merlin#Complete` in your vimrc
 - Install an auto-completer (such as YouCompleteMe) and a syntastic (such syntastic or ALE)
-- If you use vscode, you might like this extension
-
-  - [OCaml and Reason IDE](https://marketplace.visualstudio.com/items?itemName=freebroccolo.reasonml)
-  - [OCaml Platform](https://marketplace.visualstudio.com/items?itemName=ocamllabs.ocaml-platform)
+- If you use VSCode, then use [OCaml Platform](https://marketplace.visualstudio.com/items?itemName=ocamllabs.ocaml-platform)
```
Comment (mimoo) [inline README-dev.md:118]: you can delete this line now I imagine

File: Makefile [hunk 3]
```diff
@@ -80,32 +80,32 @@ check: ocaml_checks libp2p_helper
...
 build_archive_all_sigs: ocaml_checks reformat-diff
 	$(info Starting Build)
-	ulimit -s 65532 && (ulimit -n 10240 || true) && dune build src/app/archive/archive.exe src/app/archive/archive_testnet_signatures.exe src/app/archive/archive_mainnet_signatures.exe --profile=$(DUNE_PROFILE)
+	((ulimit -s 65532 || true) || true) && (ulimit -n 10240 || true) && dune build src/app/archive/archive.exe src/app/archive/archive_testnet_signatures.exe src/app/archive/archive_mainnet_signatures.exe --profile=$(DUNE_PROFILE)
 	$(info Build complete)
 
 build_rosetta: ocaml_checks
 	$(info Starting Build)
-	ulimit -s 65532 && (ulimit -n 10240 || true) && dune build src/app/archive/archive.exe src/app/rosetta/rosetta.exe src/app/rosetta/ocaml-signer/signer.exe --profile=$(DUNE_PROFILE)
+	((ulimit -s 65532 || true) || true) && (ulimit -n 10240 || true) && dune build src/app/archive/archive.exe src/app/rosetta/rosetta.exe src/app/rosetta/ocaml-signer/signer.exe --profile=$(DUNE_PROFILE)
```
Comment (mimoo) [inline Makefile:103]: double `|| true) || true)` in a few instances

**Review by mimoo — APPROVED** [review]
Nice :) !

File: Makefile (hunk 1 above)
Comment (bkase) [reply Makefile:74]: it's a relic of a very old time

**Review by jspada — APPROVED** [review]
Nice!!!   💯  Approved with minor fix requested!

**Review by lk86 — APPROVED** [review]
Providing an Infra stamp, though i'm equally worried about the double || true sequences and would prefer to see it cleaned up before merging

**Comment by querolita** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]
Amazing, great work!

**Pull request merged**